src/channel.c

```diff
@@ -1571,9 +1569,15 @@ libssh2_channel_get_exit_signal(LIBSSH2_CHANNEL *channel,
 /*
  * _libssh2_channel_receive_window_adjust
  *
- * Adjust the receive window for a channel by adjustment bytes. If the amount
- * to be adjusted is less than LIBSSH2_CHANNEL_MINADJUST and force is 0 the
- * adjustment amount will be queued for a later packet.
+ * Adjust the receive window for the channel. Adjustment is the number
+ * of bytes that we plan to read inmediately after this call. If force
+ * is false, the adjustment packed is only sent when it is bigger than
+ * LIBSSH2_CHANNEL_MINADJUST.
```

Comment:
Now, the automatic window adjustment should be enough to guarantee a proper behavior and any explicit adjustment can be safely ignored.

Though, IIRC, there where some cases where the window size could be momentarily increased as an optimization.